staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -105,7 +113,17 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	// write a new location based on the existing request pointed at the target service
 	location := &url.URL{}
 	location.Scheme = "https"
-	location.Host = handlingInfo.destinationHost
+	if r.router != nil {
+		ctx = genericapirequest.WithNamespace(ctx, handlingInfo.namespace)
+		rloc, _, err := r.router.ResourceLocation(ctx, handlingInfo.serviceName)
```

Comment:
also, making this conditional based on whether router is set makes the AddAPIService destinationHost parameter sometimes be ignored, which is really confusing